### PR TITLE
Fix cmake paths after install.

### LINKED ISSUE #6: Broken path in lib/cmake files
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

I installed `kokkos` using 
```
mamba install kokkos
```
When trying to build `kokkos-kernels` on top of it, I get an error 
```
rm -rf build && cmake -S . -B build  -DCMAKE_CXX_COMPILER=g++  -DCMAKE_INSTALL_PREFIX=/tmp/opt/kokkos
[...]
CMake Error in CMakeLists.txt:
  Imported target "Kokkos::kokkos" includes non-existent path

    "/home/conda/feedstock_root/build_artifacts/kokkos_1676604701603/_build_env/x86_64-conda-linux-gnu/sysroot/usr/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```
Looking for the broken path, I found
```
grep -rl feedstock_root /home/vincent.michaud-rioux/miniforge3 --include=*.cmake
/home/vincent.michaud-rioux/miniforge3/envs/kokkos/lib/cmake/Kokkos/KokkosConfigCommon.cmake
/home/vincent.michaud-rioux/miniforge3/envs/kokkos/lib/cmake/Kokkos/KokkosTargets.cmake
/home/vincent.michaud-rioux/miniforge3/pkgs/kokkos-3.7.01-hf52228f_0/lib/cmake/Kokkos/KokkosConfigCommon.cmake
/home/vincent.michaud-rioux/miniforge3/pkgs/kokkos-3.7.01-hf52228f_0/lib/cmake/Kokkos/KokkosTargets.cmake
```
If I replace the relevant part of the path by `/home/vincent.michaud-rioux/miniforge3/envs/kokkos` or `/home/vincent.michaud-rioux/miniforge3/pkgs/kokkos-3.7.01-hf52228f_0`, then the build goes through. 

### Installed packages

```shell
# packages in environment at /home/vincent.michaud-rioux/miniforge3/envs/kokkos:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
binutils                  2.39                 hdd6e379_1    conda-forge
binutils_impl_linux-64    2.39                 he00db2b_1    conda-forge
binutils_linux-64         2.39                h5fc0e48_11    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
c-compiler                1.5.2                h0b41bf4_0    conda-forge
ca-certificates           2022.12.7            ha878542_0    conda-forge
cmake                     3.25.3               h077f3f9_0    conda-forge
compilers                 1.5.2                ha770c72_0    conda-forge
cxx-compiler              1.5.2                hf52228f_0    conda-forge
expat                     2.5.0                h27087fc_0    conda-forge
fortran-compiler          1.5.2                hdb1a99f_0    conda-forge
gcc                       11.3.0              h02d0930_11    conda-forge
gcc_impl_linux-64         11.3.0              hab1b70f_19    conda-forge
gcc_linux-64              11.3.0              he6f903b_11    conda-forge
gfortran                  11.3.0              ha859ce3_11    conda-forge
gfortran_impl_linux-64    11.3.0              he34c6f7_19    conda-forge
gfortran_linux-64         11.3.0              h3c55166_11    conda-forge
gxx                       11.3.0              h02d0930_11    conda-forge
gxx_impl_linux-64         11.3.0              hab1b70f_19    conda-forge
gxx_linux-64              11.3.0              hc203a17_11    conda-forge
kernel-headers_linux-64   2.6.32              he073ed8_15    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
kokkos                    3.7.01               hf52228f_0    conda-forge
krb5                      1.20.1               h81ceb04_0    conda-forge
ld_impl_linux-64          2.39                 hcc3a1bd_1    conda-forge
libcurl                   7.88.1               hdc1c0ab_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libgcc-devel_linux-64     11.3.0              h210ce93_19    conda-forge
libgcc-ng                 12.2.0              h65d4601_19    conda-forge
libgfortran5              12.2.0              h337968e_19    conda-forge
libgomp                   12.2.0              h65d4601_19    conda-forge
libnghttp2                1.52.0               h61bc06f_0    conda-forge
libsanitizer              11.3.0              h239ccf8_19    conda-forge
libssh2                   1.10.0               hf14f497_3    conda-forge
libstdcxx-devel_linux-64  11.3.0              h210ce93_19    conda-forge
libstdcxx-ng              12.2.0              h46fd767_19    conda-forge
libuv                     1.44.2               h166bdaf_0    conda-forge
libzlib                   1.2.13               h166bdaf_4    conda-forge
ncurses                   6.3                  h27087fc_1    conda-forge
ninja                     1.11.1               h924138e_0    conda-forge
openssl                   3.0.8                h0b41bf4_0    conda-forge
rhash                     1.4.3                h166bdaf_0    conda-forge
sysroot_linux-64          2.12                he073ed8_15    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zlib                      1.2.13               h166bdaf_4    conda-forge
zstd                      1.5.2                h3eb15da_6    conda-forge
```

### Environment info

```shell
active environment : kokkos
    active env location : /home/vincent.michaud-rioux/miniforge3/envs/kokkos
            shell level : 2
       user config file : /home/vincent.michaud-rioux/.condarc
 populated config files : /home/vincent.michaud-rioux/miniforge3/.condarc
                          /home/vincent.michaud-rioux/.condarc
          conda version : 22.11.1
    conda-build version : 3.23.3
         python version : 3.10.8.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.31=0
                          __linux=5.14.0=0
                          __unix=0=0
       base environment : /home/vincent.michaud-rioux/miniforge3  (writable)
      conda av data dir : /home/vincent.michaud-rioux/miniforge3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/vincent.michaud-rioux/miniforge3/pkgs
                          /home/vincent.michaud-rioux/.conda/pkgs
       envs directories : /home/vincent.michaud-rioux/miniforge3/envs
                          /home/vincent.michaud-rioux/.conda/envs
               platform : linux-64
             user-agent : conda/22.11.1 requests/2.28.2 CPython/3.10.8 Linux/5.14.0-1058-oem ubuntu/20.04.5 glibc/2.31
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by vincentmr**
Resolves #6

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -11,6 +11,8 @@ ${CMAKE_ARGS} \
 -DKokkos_ENABLE_OPENMP=ON \
 -DKokkos_ENABLE_EXAMPLES=OFF \
 -DKokkos_ENABLE_SERIAL=ON \
+-DKokkos_ENABLE_LIBDL:BOOL=OFF \
+-DKokkos_ENABLE_LAUNCH_COMPILER:BOOL=OFF \
```
[the remark this answers is not in the document]
Comment (vincentmr) [reply recipe/build.sh:15]: I wanted to see whether this was related to `Kokkos_CXX_COMPILER` in
```
KokkosConfigCommon.cmake:SET(Kokkos_CXX_COMPILER "/home/conda/feedstock_root/build_artifacts/kokkos_1676604701603/_build_env/bin/x86_64-conda-linux-gnu-c++")
```
but it doesn't appear so.

[the remark this answers is not in the document]
Comment (carterbox) [reply recipe/build.sh:15]: Does the wong compiler path in Kokkos_CXX_COMPILER prevent you from using the library correctly?

[the remark this answers is not in the document]
Comment (vincentmr) [reply recipe/build.sh:15]: No you're correct. I just set to change this path everywhere, but only the `libdl` path causes issue installing kokkos-kernels, at least in my current env.

[the remark this answers is not in the document]
Comment (vincentmr) [reply recipe/build.sh:15]: So I guess we can just merge your PR and I'll post my kokkos-kernels PR in staged-recipes.

**Pull request closed without merging**